File: Creational/Abstract Factory/JavaScript/js/sedanFactory.js
import IAutoBodyFactory from './factory.js';
import { SedanLeftDoor, SedanRightDoor, SedanHood } from './product.js';

// Concrete Factory for Sedan
class SedanFactory extends IAutoBodyFactory {
    createLeftDoor() {
        return new SedanLeftDoor();
    }

    createRightDoor() {
        return new SedanRightDoor();
    }

    createHood() {
        return new SedanHood();
    }
}

export default SedanFactory;

File: Creational/Abstract Factory/C#/SedanFactory.cs
namespace AutoBodyFactory {
    public class SedanFactory : IAutoBodyFactory {
        public IAutoBodyPart CreateLeftDoor() {
            return new SedanLeftDoor();
        }

        public IAutoBodyPart CreateRightDoor() {
            return new SedanRightDoor();
        }

        public IAutoBodyPart CreateHood() {
            return new SedanHood();
        }
    }
}
